Modules/fact_manager.py

```diff
@@ -0,0 +1,337 @@
+                        f"{self._FACT_TABLE} WHERE name=%s AND lang=%s")
+
+        try:
+            result = await self.query(query, (name, lang))
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # Check for offline database, or issues with the query.
+            log.exception(f"Could not establish existence of {name}-{lang}")
+            raise
+
+        # We are only getting a single integer as a response, so we can unpack it by index.
+        # it will always return a single integer.
+        return result[0][0]
+
+    async def fact_history(self, fact_name: str, fact_lang: str) -> list:
+        """
+        Pulls the last 5 transaction logs for a fact, as a tuple
+
+        >>> history_list = self.fact_history("test", "en")
+
+        Args:
+            fact_name: Name of the fact
+            fact_lang: LangID of the fact.
+
+        Returns: tuple of transaction log items, for fact.
+        """
+        query = sql.SQL(f"SELECT name, lang, author, message, ts, old, new "
+                        f"FROM {self._FACT_LOG} WHERE name=%s AND lang=%s "
+                        f"ORDER BY ts DESC LIMIT 5")
+
+        query_data = (fact_name, fact_lang)
+
+        try:
+            result = await self.query(query, query_data)
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # ProgrammingError is a query failure, DatabaseError is database unavailable.
+            log.exception(f"Unable to retrieve history for {fact_name}-{fact_lang}")
+            raise
+
+        return result
+
+    async def find(self, name: str, lang: str) -> Fact:
+        """
+        Queries the database for a fact, and returns a populated Fact object.
+
+        See Fact class for more information on Fact properties (Modules\fact.py)
+
+        >>> example_query = sql.SQL("SELECT * some_table WHERE some_value=%s")
+        ... returned_fact = await self.query(example_query, ('something',))
+
+        Args:
+            name: name of fact to search, ie. 'prep'
+            lang: language ID for fact, defaults to 'en' (see Fact Class)
+
+        Returns: Fact()
+
+        """
+        # Build SQL Object for our query
+        query = sql.SQL(f"SELECT name, lang, message, aliases, author, edited, editedby, mfd from "
+                        f"{self._FACT_TABLE} where name=%s AND lang=%s")
+
+        # await our raw result from query
+        try:
+            rows = await self.query(query, (name, lang))
+        except (psycopg2.DatabaseError, psycopg2.ProgrammingError) as error:
+            # Check for offline database, or query errors
+            log.exception("Unable to find fact due to exception.")
+            raise
+
+        # unpack query into a fact object, or return None if there is no result.
+        return Fact(*rows[0]) if rows else None
+
+    async def log(self, fact_name: str, fact_lang: str, author: str, msg: str,
```

Comment:
I fear this name is too similar to the `LOG` field we use to store the module's logger. Consider revising.

Reply:
log has been refactored to ``add_transaction``

Fixed in 0d99f5d9.